packages/flutter/lib/src/painting/text_painter.dart

```diff
@@ -351,6 +356,22 @@ class TextPainter {
     _needsLayout = false;
   }
 
+  void layoutToMaxIntrinsicWidth({ double maxWidth }) {
+    if (!_needsLayout && width == maxIntrinsicWidth && width <= maxWidth)
```

Comment:
Can maxIntrinsicWidth ever be greater than the maxWidth? If not, I'm not sure this makes sense.